cli/cli_app.go

```diff
@@ -124,6 +127,18 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 		return err
 	}
 
+	if terragruntOptions.Source != "" {
+		if err := checkoutTerraformSource(terragruntOptions.Source, terragruntOptions); err != nil {
+			return err
+		}
+	}
+
+	if terragruntOptions.Source == "" && conf.Terraform != nil && len(conf.Terraform.Source) > 0 {
+		if err := checkoutTerraformSource(conf.Terraform.Source, terragruntOptions); err != nil {
+			return err
+		}
+	}
```

Comment:
There are two ways to tell Terragrunt that it needs to download some Terraform templates:

1. The `--terragrunt-source` command-line option.
1. A `terraform` block in `.terragrunt`. 

These if-statements pick them out in proper priority order. That said, this code was a bit convoluted for that, so I've refactored it as follows: https://github.com/gruntwork-io/terragrunt/pull/111/commits/8ecd053a8ba9d3d19d41aafc77eb0aa962238baf